pkg/allocator/allocator.go

```diff
@@ -724,7 +733,7 @@ func (a *Allocator) Allocate(ctx context.Context, key AllocatorKey) (idpool.ID,
 		}
 
 		// FIXME: Add non-locking variant
-		value, isNew, firstUse, err = a.lockedAllocate(ctx, key)
+		value, isNew, firstUse, err = a.lockedAllocate(ctx, key, attempt == a.maxAllocAttempts-1)
```

Comment:
What is the reason for using the non-seeded approach for the last attempt? I would assume that to have practically the same likelihood of hitting a conflict as any other approach, especially because at this point one of the agents should have already won the race.

Reply:
Basically, paranoia.

The specific scenario I'm concerned about is a slow k8s watch, where the agent learns about an already-existing CiliumIdentity far too late. A stale watch means a stale cache of already-allocated identities, which means we will continuously pick the same numeric ID.

We could also mark an ID as allocated on conflict, but that also risks bugs and needs to be carefully considered. Honestly, the design could use a bit of a re-think (but it works well enough now).